File: caesar/caesar.c
//Encrypts plaintext to cyphertext via a given key
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>

// Receive command line arguments
int main(int argc, string argv[])
{

    // Check input is valid number of strings
    if (argc != 2)
    {
        printf("usage : ./caesar k \n");
        return 1;
    }

    // Convert string to interger (ASCII to int)
    int k = atoi(argv[1]);

    //Prompt user for plaintext
    string p = get_string("plaintext: ");

    // Start encryption process
    printf("ciphertext: ");

    // Operate on each letter individually
    for (int i = 0; i < strlen(p); i++)
    {
        if (isalpha(p[i]))
        {
            // Check for case
            if (isupper(p[i]))
            {
                // Check to see whether 'wrap-around' is neseccary
                if (p[i] + (k % 26) <= 'Z')
                {
                    printf("%c", p[i] + (k % 26));
                }
                else
                    // Count forwards from 'A'
                {
                    printf("%c", ((p[i] + (k % 26)) - 90) + 64);
                }
            }
            else if (islower(p[i]))
            {
                // Same process for lower case
                if (p[i] + (k % 26) <= 'z')
                {
                    printf("%c", p[i] + (k % 26));
                }
                else
                {
                    printf("%c", ((p[i] + (k % 26)) - 122) + 96);
                }
            }
        }
        // If char is not alphabetical
        else
        {
            printf("%c", p[i]);
        }
    }
    // New line
    printf("\n");
    return 0;
}

File: caesar/caesar.py
from cs50 import get_string
import sys


def main():
    # Check input
    if not len(sys.argv) == 2:
        print("usage : python caesar.py k")
        exit(1)

    key = int(sys.argv[1])

    # Get plaintext
    plaintext = get_string("plaintext: ")

    # Calculate cyphertext
    ciphertext = []
    for i in range(len(plaintext)):
        ciphertext.append(determine(plaintext[i], key))
    print("ciphertext: " + ''.join(ciphertext))

    exit(0)


# Function to process char
def determine(letter, key):
    if letter.isalpha():
        if letter.isupper():
            if ord(letter) + (key % 26) <= 90:
                return chr(ord(letter) + (key % 26))
            else:
                return chr(ord(letter) + (key % 26) - 26)
        elif ord(letter) + (key % 26) <= 122:
            return chr(ord(letter) + (key % 26))
        else:
            return chr(ord(letter) + (key % 26) - 26)
    else:
        return letter


# End main
if __name__ == "__main__":
    main()